process/tests/test_lock.py

```diff
@@ -0,0 +1,85 @@
+import threading
+import time
+
+import pytest
+
+from datadog_checks.process.process import ReadWriteLock
+
+READER_SLEEP = 0.005
+
+
+@pytest.mark.unit
+def test_read_lock():
```

Comment:
Not sure what to think of this test, since it breaks encapsulation. Should the reader count be exposed instead?

Reply:
I think it's fine, unless you need to have more tests like this.